### ISSUE TITLE
Wrong dates when export has format dd/MM/YYYY

### ISSUE THREAD

**Issue by mlomb** [Pustur/whatsapp-chat-parser#237]
Parsing  the following export:
```
30/12/2020 13:00 - User: text
13/1/2021 13:00 - User: text
```
Produces the following output:
```json
[
    {
        "date": "2020-12-30T16:00:00.000Z",
        "author": "User",
        "message": "text"
    },
    {
        "date": "2022-01-01T16:00:00.000Z",
        "author": "User",
        "message": "text"
    }
]
```
Note the second date.

I didn't check the root of the problem, but it seems a `dd/MM` / `MM/dd` problem.

And it seems that after that date is broken, every following date is broken too 😢

**Comment by Pustur** [comment]
Thanks for the bug report, 

I can reproduce the issue and I'm going to investigate tomorrow.  
Seems like the first date is parsed correctly, where the day comes before the month, but in the second date it gets it wrong and thinks that the day comes after the month.  
Weird since it should always stick to one or the other throughout...

**Comment by Pustur** [comment]
Hey @mlomb, thank you so much for bringing this issue to my attention. 

It was a pretty important bug in the code that would sort the date numbers erroneously in specific cases like yours.

I released the fix as version `3.2.1`, let me know if you find any new problems.
